test/extensions/resource_monitors/cgroup_memory/cgroup_memory_monitor_test.cc

```diff
@@ -0,0 +1,251 @@
+#include "source/extensions/resource_monitors/cgroup_memory/cgroup_memory_monitor.h"
+#include "source/extensions/resource_monitors/cgroup_memory/cgroup_memory_paths.h"
+#include "source/extensions/resource_monitors/cgroup_memory/cgroup_memory_stats_reader.h"
+
+#include "test/mocks/filesystem/mocks.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::NiceMock;
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace CgroupMemory {
+namespace {
+
+class ResourcePressure : public Server::ResourceUpdateCallbacks {
```

Comment:
Not a blocker, there a couple resource monitor tests that use this same abstraction -- having it as a common test class would be useful to de-dup.